File: src/main/java/calculator/Calculator.java
package calculator;

public class Calculator {

    int valueDisplayed;

    public Calculator() {
        this.valueDisplayed = 0;
}

    public Calculator(int initialValue) {
        this.valueDisplayed = initialValue;
    }

    public void add(int x) {
        this.valueDisplayed += x;
    }

    public void subtract(int x) {
        this.valueDisplayed -= x;
    }

    public void power(int x) {
        this.valueDisplayed = (int) Math.pow(this.valueDisplayed, x);
    }

    public int getResult() {
        return this.valueDisplayed;
    }

    public boolean tryToSetDisplay(int x, boolean canChangeDisplay) {
        if(canChangeDisplay) {
            setDisplay(x);
            return true;
        } else {
            return false;
        }
    }


    private void setDisplay(int x)
    {
        this.valueDisplayed = x;
    }

}

File: src/main/java/calculator/Calculator.kt
package calculator

import kotlin.math.pow

class Calculator(var displayedValue: Double = 0.0) {

    fun add(x: Double) {
        displayedValue += x
    }

    fun subtract(x: Double) {
        displayedValue -= x
    }

    fun power(x: Double) {
        displayedValue = displayedValue.pow(x)
    }

    fun getResult() = displayedValue

    fun tryToSetDisplay(x: Double, canChangeDisplay: Boolean): Boolean {
        return if (canChangeDisplay) {
            setDisplay(x)
            true
        } else {
            false
        }
    }

    private fun setDisplay(x: Double) {
        displayedValue = x
    }
}

